### ISSUE TITLE
DelayedJob Configuration

### ISSUE THREAD

**Issue by hennevogel** [openSUSE/open-build-service#3342]
There are several problems with our setup of [delayed_job](https://github.com/collectiveidea/delayed_job) setup

- it doesn't handle failed jobs at all, it just continuously tries to execute them
- it doesn't limit execution time, jobs can run forever
- it mostly uses one queue

In combination with our [clockworkd](http://www.rubydoc.info/gems/clockworkd/0.2.5) setup which just adds job, after job, after job on top of the delayed job queue (#3341) we run into serious problems.

This setup needs some serious thought.

**Comment by evanrolfe** [comment]
@hennevogel maybe worth discussing with this card in grooming today:
https://trello.com/c/iJsZmbnZ/378-improve-replace-existing-delayed-job-system

**Comment by hennevogel** [comment]
In the meanwhile I found defaults in the [DJ code](https://github.com/collectiveidea/delayed_job/blob/ccbde3dbcd90a409fe3dceae906fa6292ef09692/lib/delayed/worker.rb#L11)

```
DEFAULT_MAX_ATTEMPTS     = 25
DEFAULT_MAX_RUN_TIME     = 4.hours
```

**Comment by evanrolfe** [comment]
> it doesn't handle failed jobs at all, it just continuously tries to execute them

I dont think thats the behaviour of DJ but the events implementation from before? This no longer happens with events anyway.

> it doesn't limit execution time, jobs can run forever
Max run time is 4 hours as you just said.

> it mostly uses one queue

We have 6 different queues.

I would suggest closing this issue. The points dont really make sense to me?

**Comment by hennevogel** [comment]
>>  it mostly uses one queue

> We have 6 different queues.

It's not about the number of queues we have. The majority of our jobs happen in `quick` and the second most busy queue is only by accident `default`.

**Comment by hennevogel** [comment]
We handled this in various pull requests now...

**Comment by bgeuken** [comment]
@openSUSE/open-build-service This still seems to be an issue:

http://errbit.suse.de/apps/53bac9aad9d1af372d012a0c/problems/59aec6052bfc42072129f5af

**Comment by hennevogel** [comment]
Your error was #3587, addressed in #3874
